### PR TITLE
Update for React 15 and React-portal 3

### PR DESCRIPTION
**Pull request by paulmelnikow**
Thank you for this useful library! Your projects look really interesting too!

I updated it to work with React 15. The pinned version of react-portal doesn't work so I updated it. In my experience, React dependencies are easier to manage at the top level, so I made react-portal a peer dependency, like react was.

I made some other changes to the package.json to get the project to import correctly.

I tried to run the linter on a clean install, but it seems there are some missing dependencies. I tried upgrading them all, but ended up with a slough of lint errors as a result. I might need your help with that.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,20 +13,31 @@
...
     "react-component-gulp-tasks": "^0.7.6",
-    "react-dom": "^0.14.0"
+    "react-dom": "^15.0.0",
+    "react-portal": "^3.0.0"
   },
   "peerDependencies": {
-    "react": "^0.14.0"
+    "react": ">=15.0.0",
+    "react-portal": ">=3.0.0"
+  },
```
Comment (paulmelnikow) [inline package.json]: The changes below were made to get the module to import correctly from a parent project via browserify. The `babel` settings might be required for webpack too.

File: src/ReactEditableSvgLabel.js [hunk 2]
```diff
@@ -52,10 +52,16 @@ var ReactEditableSvgLabel = React.createClass({
   },
 
   render () {
-    var label = <text ref='label' {...this.props}>{this.props.children}</text>;
+    // Omit onChange, minLabelWidth, and children.
+    var passThroughProps = Object.assign({}, this.props);
+    Object.keys(this.constructor.propTypes).forEach(function (key) {
+      delete passThroughProps[key];
+    });
```
Comment (paulmelnikow) [inline src/ReactEditableSvgLabel.js]: This change avoids a warning `Unknown prop `minLabelWidth` on <text> tag.` The idea behind this change is that the component should pass through additional props to the `text` element, but not its own props.

File: src/ReactEditableSvgLabel.js [hunk 3]
```diff
@@ -52,10 +52,16 @@ var ReactEditableSvgLabel = React.createClass({
...
+
+    var label = <text ref='label' {...passThroughProps}>{this.props.children}</text>;
     return (
-      <Portal openByClickOn={label} closeOnOutsideClick style={{position: 'absolute', top: this.state.labelY, left: this.state.labelX}}>
-        <input type='text' value={this.props.children} onChange={this.handleChangeText} style={{width: Math.max(this.props.minLabelWidth, this.state.labelWidth), height: this.state.labelHeight}} />
+      <Portal openByClickOn={label} closeOnOutsideClick>
+        <input type='text' value={this.props.children} onChange={this.handleChangeText} style={{position: 'absolute', top: this.state.labelY, left: this.state.labelX, width: Math.max(this.props.minLabelWidth, this.state.labelWidth), height: this.state.labelHeight}} />
```
Comment (paulmelnikow) [inline src/ReactEditableSvgLabel.js]: It seems this version of `Portal` ignores the `style` prop. I tried adding a div (within the portal), which worked, but adding the style directly to the `input` worked just as well.

**Comment by mathisonian** [comment]
Awesome, thanks @paulmelnikow! I'm traveling now but will look closely and merge early next week. 

One note, I think it probably makes sense export the compiled `lib` folder via `main`

**Comment by paulmelnikow** [comment]
Sounds good. Makes sense about the export. Have updated that.

**Comment by mathisonian** [comment]
Thanks @paulmelnikow! I've merged and published  version `1.1.0` to npm including this change and #4. I've updated the linting script to use `semistandard`. I changed the `focusOnOpen` prop to be true by default since I think this is a very reasonable behavior. 

I've also added you as a collaborator since you've made a substantial contribution

**Comment by paulmelnikow** [comment]
Awesome, thanks for reviewing everything. Thanks for adding me to the project too!

I was able to run the linter, though I can't quite get the project to work correctly via npm. I'm tracking down the problem now.

**Comment by mathisonian** [comment]
Ah - i think I forgot to push the `lib/` folder to npm - I'll push a patch

**Comment by mathisonian** [comment]
Seems to be working now. I was able to use it in a local project by installing from npm

**Comment by paulmelnikow** [comment]
Yup, all good. Thanks!

**Pull request merged**